k8s/plugin/src/main.rs

```diff
@@ -14,19 +15,40 @@ struct CliArgs {
     #[clap(subcommand)]
     operations: Operations,
 
-    /// Kubernetes namespace of mayastor service
-    #[clap(global = true, long, short = 'n', default_value = "mayastor")]
-    namespace: String,
+    /// Kubernetes namespace of mayastor service [default: mayastor].
+    #[clap(global = true, long, short = 'n')]
+    namespace: Option<String>,
 
     #[clap(flatten)]
     args: resources::CliArgs,
+
+    /// Use namespace from the current `kubeconfig` context.
+    #[clap(global = true, long, hide = true, default_value = "false")]
+    namespace_from_context: bool,
 }
 
 impl CliArgs {
-    fn args() -> Self {
+    async fn args() -> Result<Self, anyhow::Error> {
         let mut args = CliArgs::parse();
-        args.args.namespace = args.namespace.clone();
-        args
+        args.args.namespace = if let Some(namespace) = &args.namespace {
+            namespace.to_string()
+        } else if args.namespace_from_context {
+            let client = if let Some(path) = &args.args.kube_config_path {
+                kube_proxy::client_from_kubeconfig(Some(path.clone()))
+                    .await
+                    .map_err(|_| {
+                        anyhow::anyhow!("Couldnt create client from supplied kubeconfig")
+                    })?
+            } else {
+                Client::try_default()
+                    .await
+                    .map_err(|_| anyhow::anyhow!("Couldnt create client from default kubeconfig"))?
+            };
```

Comment:
Can't we just do this?

```suggestion
            let client = kube_proxy::client_from_kubeconfig(args.args.kube_config_path.clone())
                    .await
                    .map_err(|_| {
                        anyhow::anyhow!("Couldn't create client from supplied kubeconfig")
                    })?;
```

Reply:
Yes should work.